src/Utilities/ProtocolHelpers.hpp

```diff
@@ -7,11 +7,22 @@
 
 #include "Utilities/TypeTraits.hpp"
 
+/*!
```

Comment:
Yes my intention is that protocols are placed in `protocols` namespaces (just like our `Tags` namespaces). This is written at the end of the overview section in the docs.

The `Utilities/ProtocolHelpers.hpp` defines utility metafunctions that are not protocols. I suppose we could place them in a top-level `protocols` namespace, but that could be misleading. E.g. the following does not read very nicely:
```cpp
struct Person : protocols::ConformsTo<protocols::Named> {};

static_assert(protocols::conforms_to<Person, protocols::Named>, "...");
```
A lot of the `Utilities` are in the global namespace, and that's probably fine because they supply general functionality that could be in the standard library, as opposed to things in `Domain` that should really be in a namespace. So the code above then reads:
```cpp
struct Person : ConformsTo<protocols::Named> {};

static_assert(conforms_to<Person, protocols::Named>, "...");
```
What do you think?

Reply:
I think this needs to go in a namespace... Personally I think we should've written all of spectre inside a `spectre` namespace with appropriate nested namespaces but here we are...

Reply:
Ok, how about something like a top-level `prtcl` namespace? Or use `tt`?

Reply:
or `util`?

Reply:
I'd just go with protocols or protos or proto. I don't love `util` because that's rather vague. I would find it easier if all the protocols stuff is in a `protocols` namespace, then I don't have to think about which namespace things are in when. The other option is to put all the protocols into the `tt` namespace, since now they really are just type traits

Reply:
👍 I moved the protocol helpers into the top-level `tt` namespace. Please check the fixup